lib/inspec/profile.rb

```diff
@@ -89,7 +89,7 @@ def initialize(source_reader, options = {})
       @writable = options[:writable] || false
       @profile_id = options[:id]
       @cache = options[:cache] || Cache.new
-      @backend = options[:backend] || Inspec::Backend.create(options)
+      @backend = options[:backend] || Inspec::Backend.create(options.select { |k, _| k != 'target' })
```

Comment:
I think we need to make the separation more clear. The problem with `the options we do need` is that the backend is getting most of the cli options, since all the ssh or winrm options are just passed through. I think we should create an issue to solve that in another case. The real issue I am seeing here is that we need to run integration tests with Chef Automate for every PR, since a change should break the build.

So we need to do two things:
- Integration testing with Chef Compliance / Chef Automate
- Make the interface between inspec / train more clear